runtime/src/vm/memory.rs

```diff
@@ -16,12 +16,12 @@ pub struct HandlerMemory {
   /// Global memory reference
   gmem: &'static Vec<u8>,
   /// Memory of the handler for fixed size data types
-  mem: Vec<i64>,
+  pub mem: Vec<i64>,
```

Comment:
Do we want to go back to making these private at some point or we are ok making all of the memory fields public?

Reply:
I personally prefer having these fields public. There are a couple of methods in here that support only a single opcode because that logic couldn't live in the opcode with the struct fields being private. That causes `HandlerMemory` to have to house logic that it itself "should care about" and makes the separation of concerns blurry.

Of course, public fields can also cause issues in the other direction, which is why I'm not pushing for making everything public. All of the logic involving global memory addresses makes fields involving that pretty convoluted and something the opcodes should *not* need to deal with.

Reply:
I agree that global memory should not be public, it is just becoming harder to understand where abstraction is drawn. e.g. what should be public and what shouldn't.

Reply:
True. I want to do an RFC at some point in the future (after launch) where we go back and streamline the memory model with both performance and clarity. Similar to the "remove ANTLR from the compiler" work that's nice to have but doesn't block functionality.

Reply:
💯 agree!